lib/cgo/libsky_handle.go

```diff
@@ -91,14 +136,406 @@ func lookupConfigHandle(handle C.Config__Handle) (*cli.Config, bool) {
+}
+
+func lookupWalletNotesHandle(handle C.WalletNotes_Handle) (*wallet.Notes, bool) {
+	obj, ok := lookupHandle(C.Handle(handle))
+	if ok {
+		if obj, isOK := (obj).(*wallet.Notes); isOK {
+			return obj, true
+		}
+	}
+	return nil, false
+}
+
+func registerWalletReadableNotesHandle(obj *wallet.ReadableNotes) C.WalletReadableNotes_Handle {
+	return (C.WalletReadableNotes_Handle)(registerHandle(obj))
+}
+
+func lookupWalletReadableNotesHandle(handle C.WalletReadableNotes_Handle) (*wallet.ReadableNotes, bool) {
+	obj, ok := lookupHandle(C.Handle(handle))
+	if ok {
+		if obj, isOK := (obj).(*wallet.ReadableNotes); isOK {
+			return obj, true
+		}
+	}
+	return nil, false
+}
+
+func registerOutputsResultHandle(obj *webrpc.OutputsResult) C.OutputsResult_Handle {
+	return (C.OutputsResult_Handle)(registerHandle(obj))
+}
+
+func lookupOutputsResultHandle(handle C.OutputsResult_Handle) (*webrpc.OutputsResult, bool) {
+	obj, ok := lookupHandle(C.Handle(handle))
+	if ok {
+		if obj, isOK := (obj).(*webrpc.OutputsResult); isOK {
+			return obj, true
+		}
+	}
+	return nil, false
+}
+
+func registerStatusResultHandle(obj *webrpc.StatusResult) C.StatusResult_Handle {
+	return (C.StatusResult_Handle)(registerHandle(obj))
+}
+
+func lookupStatusResultHandle(handle C.StatusResult_Handle) (*webrpc.StatusResult, bool) {
+	obj, ok := lookupHandle(C.Handle(handle))
+	if ok {
+		if obj, isOK := (obj).(*webrpc.StatusResult); isOK {
+			return obj, true
+		}
+	}
+	return nil, false
+}
+
+func registerAddressUxOutHandle(obj *coin.AddressUxOuts) C.AddressUxOuts_Handle {
+	return (C.AddressUxOuts_Handle)(registerHandle(obj))
+}
+
+func lookupAddressUxOutHandle(handle C.AddressUxOuts_Handle) (*coin.AddressUxOuts, bool) {
+	obj, ok := lookupHandle(C.Handle(handle))
+	if ok {
+		if obj, isOK := (obj).(*coin.AddressUxOuts); isOK {
+			return obj, true
+		}
+	}
+	return nil, false
+}
+
+func registerBuildInfoHandle(obj *visor.BuildInfo) C.BuildInfo_Handle {
+	return (C.BuildInfo_Handle)(registerHandle(obj))
+}
+
+func lookupBuildInfoHandle(handle C.BuildInfo_Handle) (*visor.BuildInfo, bool) {
+	obj, ok := lookupHandle(C.Handle(handle))
+	if ok {
+		if obj, isOK := (obj).(*visor.BuildInfo); isOK {
+			return obj, true
+		}
+	}
+	return nil, false
+}
+
+func registerHashHandle(obj *hash.Hash) C.Hash_Handle {
+	return (C.Hash_Handle)(registerHandle(obj))
+}
+
+func lookupHashHandle(handle C.Hash_Handle) (*hash.Hash, bool) {
+	obj, ok := lookupHandle(C.Handle(handle))
+	if ok {
+		if obj, isOK := (obj).(*hash.Hash); isOK {
+			return obj, true
+		}
+	}
+	return nil, false
+}
+
+func registerNumberHandle(obj *secp256k1go2.Number) C.Number_Handle {
+	return (C.Number_Handle)(registerHandle(obj))
+}
+
+func lookupNumberHandle(handle C.Number_Handle) (*secp256k1go2.Number, bool) {
+	obj, ok := lookupHandle(C.Handle(handle))
+	if ok {
+		if obj, isOK := (obj).(*secp256k1go2.Number); isOK {
+			return obj, true
+		}
+	}
+	return nil, false
+}
+
```

Comment:
In here a `SKY_handle_Duplicate` function is needed . It should create a new handle pointing at the same object that the original handle .